File: GielinorSimulator/ServerApp/Model/Kingdom.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace GielinorSimulator.Model
{
    public class Kingdom : Entity
    {
        public string Government { get; set; }
        public string Monarch { get; set; }
        public string Usurper { get; set; }
        public string Capital { get; set; }
        public string Religion { get; set; }
        public string Demonym { get; set; }
        [NotMapped]
        public Date Established { get; set; }
        public int EstablishedNumber
        {
            get
            {
                if (Established != null)
                {
                    return Established.Number;
                }
                return 0;
            }
            set
            {
                if (value == 0)
                {
                    return;
                }
                Established = new Date(value);
            }
        }

        public Kingdom()
        {
            Capital = "";
            Religion = "";
            Demonym = "";
        }

        public override string ToString()
        {
            return "Kingdom String";
        }
    }
}

File: GielinorSimulator/ClientApp/src/Model/Kingdom.ts
import { Entity } from "./Entity";
import { Date } from "./Date";

interface KingdomInterface {
    Environment: string,
    Name: string,
    Government: string,
    Monarch: string,
    Usurper: string,
    Council: string,
    Capital: string,
    Religion: string,
    Demonym: string,
    Established: Date | undefined,
    Fragmented: Date | undefined,
    Reorganized: Date | undefined,
    Dissolved: Date | undefined,
    Restored: Date | undefined,
}

export class Kingdom extends Entity {
    Environment: string;
    Name: string;
    Government: string;
    Monarch: string;
    Usurper: string;
    Council: string;
    Capital: string;
    Religion: string;
    Demonym: string;
    Established: Date | undefined;
    Fragmented: Date | undefined;
    Reorganized: Date | undefined;
    Dissolved: Date | undefined;
    Restored: Date | undefined;

    constructor();
    constructor(initData?: KingdomInterface);
    constructor(private initData?: KingdomInterface) {
        super();
        this.Environment = "";
        this.Name = initData?.Name || "";
        this.Government = initData?.Government || "";
        this.Monarch = initData?.Monarch || "";
        this.Usurper = initData?.Usurper || "";
        this.Council = initData?.Council || "";
        this.Capital = initData?.Capital || "";
        this.Religion = initData?.Religion || "";
        this.Demonym = initData?.Demonym || "";
        this.Established = initData?.Established || undefined;
        this.Fragmented = initData?.Fragmented || undefined;
        this.Reorganized = initData?.Reorganized || undefined;
        this.Dissolved = initData?.Dissolved || undefined;
        this.Restored = initData?.Restored || undefined;
    }
}

export default Kingdom;